airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/report_streams.py

```diff
@@ -355,6 +356,9 @@ def _init_reports(self, profile: Profile, report_date: str) -> List[ReportInfo]:
         """
         report_infos = []
         for record_type, metrics in self.metrics_map.items():
+            if len(self._report_record_types) > 0 and record_type not in self._report_record_types:
```

Comment:
```suggestion
            if record_type not in self._report_record_types:
```

When `self._report_record_types` is an empty array, which is by default, `not in` will return `True` for any value.

Reply:
it's on `and` condition. If array is empty, the complete condition would always be false

Reply:
```python
>>> report_record_types = []
>>> len(report_record_types) > 0 and 'a_record_type' in report_record_types
False
>>> 'a_record_type' in report_record_types
False
```

I made a typo, but my point is that this if statement can be simplified.